### DISCUSSION TITLE
Keychain ACL deprecations -- what's up?

### DISCUSSION THREAD

**Discussion by sbeitzel** [chucker/Mastonaut#28] [Q&A] [answered]
In the CoreTootin module, the `KeychainController` calls down to some helper methods defined in the `NSBundle+KeychainAccess` extension. Ultimately, we are reading and writing data to the keychain and we're sticking a custom access control list on the items (if we can) so that both Mastonaut and the Quick Toot applications can access them. But the methods for defining that access control list are all deprecated. I took a quick look at the developer documentation, and it seems that *all* of the ACL methods are deprecated.

Now, I'm not too familiar with this aspect of the keychain; has anyone got an understanding of *why* these things are all deprecated? Is it just that Apple wants everyone to run this kind of sharing through iCloud (which seems kind of slimy) or is there some different way it's supposed to be done?

**Comment by sbeitzel** [comment, marked as answer]
Aha, it seems that the existing code is specifically targeting the file based keychain, while Apple is recommending using a newer way. It also looks as though this migration has already started in Mastonaut, with defining the app group. What remains, I suppose, is to stop writing to the file keychain and to read from data first and only fall back to file in the case where the data item is missing. Then, in a future version, remove all the file based keychain access altogether.

[Apple tech note](https://developer.apple.com/documentation/technotes/tn3137-on-mac-keychains)

**Reply by chucker** [reply]
Rewriting this _may_ also fix #90.
